pkg/client/clientset/versioned/typed/nodepoollabelset/v1alpha1/nodepoollabelset.go

```diff
@@ -66,7 +68,7 @@ func (c *nodePoolLabelSets) Get(name string, options v1.GetOptions) (result *v1a
 		Resource("nodepoollabelsets").
 		Name(name).
 		VersionedParams(&options, scheme.ParameterCodec).
-		Do().
+		Do(context.TODO()).
```

Comment:
**Recommended**: we **SHOULD** use `context.Background()` for requests without timeouts, because `context.TODO()` implies we it is a temporary development only dummy value, background implies it is intended to not have a timeout.

Also a reasonable timeout might be a better idea than using background.
Without thinking much unless we expect hundreds of objects I would say 30-60 seconds are reasonable, but we have to think about customer use-cases as well.

And of course, same for all other occurrences.

Reply:
Thanks for the suggestion!

Reply:
Apologies, on a second thought, it is probably best to pass down the context from above and only define it once on top level which could be user supplied value or a reasonable default (30-60s unless hundreds of objects) or background.